ale/transformation.py

```diff
@@ -1,194 +1,99 @@
+                     of frame rotations in the frame chain
     """
+    def __init__(self, *args, frame_changes = [], ephemeris_time=[], **kwargs):
+        super(FrameChain, self).__init__(*args, **kwargs)
 
-    def __init__(self, id, parent=None, rotation=None):
-        """
-        Construct a frame node with or without a parent. If a parent is specified
-        then a rotation from this frame to the parent node's frame must be
-        specified and visa-versa.
-
-        Parameters
-        ----------
-        id : int
-             The NAIF ID code for the frame
-        parent : FrameNode
-                 The parent node in the frame chain
-        rotation : ConstantRotation or TimeDependentRotation
-                   The rotation from this frame to the frame of the parent node
-        """
+        times = np.array(ephemeris_time)
+        quats = np.zeros((len(times), 4))
 
-        if (parent is None) != (rotation is None):
-            raise TypeError("parent and rotation must both be entered or both be None.")
-
-        self.children = []
-        self.id = id
-        if parent is not None:
-            self.parent = parent
-        if rotation is not None:
-            self.rotation = rotation
-
-    def __repr__(self):
-        return f'ID:{self.id}\nChildren:{self.children}'
-
-    def __del__(self):
-        """
-        Custom deletor for a FrameNode. The child node is always responsible
-        for updating the parent nodes. So this removes this node from the
-        children of its parent node.
-        """
-        if self.parent is not None:
-            self.parent.children.remove(self)
-
-    @property
-    def parent(self):
-        """
-        The parent node of this node. Returns None if this is a root node.
-        """
-        if hasattr(self, '_parent'):
-            return self._parent
-        else:
-            return None
-
-    @parent.setter
-    def parent(self, new_parent):
-        """
-        Sets a new parent node. The child node is always responsible for
-        updating the parent node. So, this removes this node from the children
-        of the old parent and adds it to the children of the new parent.
-        """
-        if self.parent is not None:
-            self._parent.children.remove(self)
-
-        new_parent.children.append(self)
-        self._parent = new_parent
-
-    def parent_nodes(self):
-        """
-        Returns the ordered list of parents starting with this node going to
-        the root node.
-        """
-        chain = [self]
-        current_parent = self.parent
-        while current_parent is not None:
-            chain.append(current_parent)
-            current_parent = current_parent.parent
-        return chain
+        for s, d in frame_changes:
+            for i, time in enumerate(times):
+                rotation_matrix = spice.pxform(spice.frmnam(s), spice.frmnam(d), time)
```

Comment:
We cannot hard code a SPICE call into this class. We need to use this same class to handle rotation derived from things like ISIS spiceinit'd cubes also.

Reply:
Handled in a new classmethod called from_spice